js/oceanex.js

```diff
@@ -0,0 +1,323 @@
+            'last': this.safeFloat (ticker, 'last'),
+            'previousClose': undefined,
+            'change': undefined,
+            'percentage': undefined,
+            'average': undefined,
+            'baseVolume': this.safeFloat (ticker, 'volume'),
+            'quoteVolume': undefined,
+            'info': ticker,
+        };
+    }
+
+    async fetchOrderBook (symbol, limit = undefined, params = {}) {
+        let response = await this.publicGetOrderBook (this.extend ({
+            'market': symbol,
+            'limit': limit,
+        }, params));
+        const orderbook = this.safeValue (response, 'data');
+        return this.parseOrderBook (orderbook, orderbook['timestamp']);
+    }
+
+    async fetchTrades (symbol, since = undefined, limit = undefined, params = {}) {
+        let response = await this.publicGetTrades (this.extend ({
+            'market': symbol,
+            'limit': limit,
+        }, params));
+        let data = this.safeValue (response, 'data');
+        return this.parseTrades (data, symbol, since, limit);
+    }
+
+    parseTrade (trade, market = undefined) {
+        let side = undefined;
+        if (trade['side'] === 'bid') {
+            side = 'buy';
+        } else if (trade['side'] === 'ask') {
+            side = 'sell';
+        }
+        return {
+            'info': trade,
+            'timestamp': trade['created_on'],
+            'datetime': trade['created_at'],
+            'symbol': trade['market'],
+            'id': trade['id'],
+            'order': undefined,
+            'type': 'limit',
+            'takerOrMaker': undefined,
+            'side': side,
+            'price': this.safeFloat (trade, 'price'),
+            'amount': this.safeFloat (trade, 'volume'),
+            'cost': undefined,
+            'fee': undefined,
+        };
+    }
+
+    async fetchTime (params = {}) {
+        let response = await this.publicGetTimestamp ();
+        return response['data'];
+    }
+
+    async fetchTradingFees (params = {}) {
+        let response = await this.publicGetFeesTrading ();
+        return response['data'];
+    }
+
+    async fetchKey (params = {}) {
+        // private method input value {'user_jwt': encrypted_jwt_token}
+        // jwt_token should be RS256 encoded
+        let response = await this.privateGetKey (this.extend (params));
+        return response['data'];
+    }
+
+    async fetchBalance (params = {}) {
+        let response = await this.privateGetMembersMe (this.extend (params));
+        const balances = this.safeValue (response['data'], 'accounts');
+        let result = { 'info': balances };
+        for (let i = 0; i < balances.length; i++) {
+            const balance = balances[i];
+            let currency = balance['currency'];
+            let account = this.account ();
+            let free = this.safeFloat (balance, 'balance');
+            let used = this.safeFloat (balance, 'locked');
+            let total = free + used;
```

Comment:
Unfortunately, this line violates the following guidelines on summing in a cross-language portable way: https://github.com/ccxt/ccxt/blob/master/CONTRIBUTING.md#adding-numbers-and-concatenating-strings

We will have to edit it to make it compatible with JS/Py/PHP. In JS and Python that does work, but, unfortunately, not in PHP.